### DISCUSSION TITLE
使用自定义标签库提示错误

### DISCUSSION THREAD

**Discussion by hulang** [top-think/framework#3061] [Q&A] [answered]
RT
@big-dream 
帮忙看看
https://github.com/top-think/think-template/issues/29

**Comment by big-dream** [comment]
不会

**Comment by hulang** [comment]
@big-dream 好吧。。。那看大佬你写了很多基于`League\Flysystem`的驱动。。
我想问个，如何自定义驱动
我自建了一个img的域名，如何使用`League\Flysystem`上传到我自建的域名中？

**Comment by big-dream** [comment]
`League\Flysystem` 主要是定义类接口规范，怎么上传得看实际情况

**Comment by augushong** [comment, marked as answer]
https://gitee.com/ulthon/ulthon_admin/blob/master/extend/think/filesystem/driver/Qiniu.php



看这个，七牛腾讯阿里


基本思路
https://www.kancloud.cn/augushong/write_by_augushong/1188527

**Comment by augushong** [comment]
而且这个框架里，实现了不同驱动补全不同的对应的前缀域名

![QQ_1725946526634](https://github.com/user-attachments/assets/6e8dc7a5-39c8-41cc-83da-d46eb96da980)

**Comment by augushong** [comment]
不过当然，这个只是把Flysystem的驱动按照tp的机制介入进来，并不是自定义flysystem驱动，但官方的驱动应该能覆盖绝大多数系统了。

**Comment by hulang** [comment]
@augushong 大佬好啊。。很久没有看到你了。

**Comment by hulang** [comment]
@augushong 大佬：`QiniuAdapter`这个类我应该如何自定义呢？就是这个自定义驱动类，我不会写额。。

**Reply by augushong** [reply]
![QQ_1725947280580](https://github.com/user-attachments/assets/686eafe4-a34d-464f-8f06-12be1a2a181d)

overtrue/flysystem-qiniu

这个是别的大佬实现的flysystem社区驱动，去社区搜一下找合适的驱动就行了

![QQ_1725947348074](https://github.com/user-attachments/assets/9ca4c739-cca1-42a6-874f-6f9220b07d2b)
